src/main/scala/zio/prelude/Associative.scala

```diff
@@ -130,6 +130,38 @@ object Associative extends Lawful[AssociativeEqual] {
       def inverse(l: => Sum[Boolean], r: => Sum[Boolean]): Sum[Boolean] = Sum(l && !r)
     }
 
+  /**
+   * The `Commutative`, `Idempotent` instance for the max of `BigDecimal` values
+   */
+  implicit val BigDecimalMaxIdempotent: Commutative[Max[BigDecimal]] with Idempotent[Max[BigDecimal]] =
```

Comment:
This could be `BigDecimalMaxCommutativeIdempotent` to reflect the commutativity in the name as well.